pallets/scoring/src/lib.rs

```diff
@@ -0,0 +1,347 @@
+		InvalidIdentifierLength,
+		// Invalid digest
+		InvalidDigest,
+		// Invalid creator signature
+		InvalidSignature,
+		// Invalid Rating Identifier
+		InvalidRatingIdentifier,
+		// Transaction already rated
+		TransactionAlreadyRated,
+		// Invalid rating value - should be between 1 and 50
+		InvalidRatingValue,
+		// Exceeds the maximum allowed entries in a single transaction
+		TooManyJournalEntries,
+		// Invalid entity signature
+		InvalidEntitySignature,
+		//Stream digest is not unique
+		DigestAlreadyAnchored,
+		//Count entry Greater than storage count
+		CountGreaterThanStorage,
+		//If Rating and count value Zero
+		CountCannotBeZero,
+		//If Rating and count value Zero
+		RatingCannotBeZero,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		///
+		/// Create a new rating identifier and associates it with its
+		/// controller. The controller (issuer) is the owner of the identifier.
+		///
+		/// * origin: the identity of the Transaction Author. Transaction author
+		///   pays the transaction fees
+		/// * tx_journal: the incoming rating entry.
+		/// * `authorization`: The authorization ID of the delegate who is
+		///   allowed to perform this action.
+		#[pallet::call_index(0)]
+		#[pallet::weight({0})]
+		pub fn entries(
+			origin: OriginFor<T>,
+			journal: RatingInputOf<T>,
+			authorization: AuthorizationIdOf,
+		) -> DispatchResult {
+			let author = <T as Config>::EnsureOrigin::ensure_origin(origin)?.subject();
+
+			let registry_id =
+				pallet_registry::Pallet::<T>::is_a_delegate(&authorization, author.clone(), None)
+					.map_err(<pallet_registry::Error<T>>::from)?;
+
+			ensure!(
+				journal.entry.rating >= T::MinScoreValue::get(),
+				Error::<T>::InvalidRatingValue
+			);
+			ensure!(
+				!<JournalHashes<T>>::contains_key(journal.digest),
+				Error::<T>::DigestAlreadyAnchored
+			);
+
+			// let id_digest = <T as frame_system::Config>::Hashing::hash(
+			// 	&[&(journal.digest).encode()[..]].concat()[..],
+			// );
+
+			let identifier = Ss58Identifier::to_scoring_id(&journal.digest.encode()[..])
+				.map_err(|_| Error::<T>::InvalidIdentifierLength)?;
+
+			ensure!(
+				!<TidEntries<T>>::contains_key(&journal.entry.tid, &journal.entry.rating_type),
+				Error::<T>::TransactionAlreadyRated
+			);
+
+			let block_number = frame_system::Pallet::<T>::block_number();
+
+			Self::aggregate_score(&journal.entry)?;
+
+			<Journal<T>>::insert(
+				&identifier,
+				&journal.entry.rating_type,
+				RatingEntryOf::<T> {
+					entry: journal.entry.clone(),
+					digest: journal.digest,
+					created_at: block_number,
+					registry: registry_id,
+					creator: author.clone(),
+				},
+			);
+			<JournalHashes<T>>::insert(journal.digest, ());
+			<TidEntries<T>>::insert(
+				&journal.entry.tid,
+				&journal.entry.rating_type,
+				&journal.entry.entity,
+			);
+
+			Self::deposit_event(Event::JournalEntry {
+				identifier,
+				entity: journal.entry.entity.clone(),
+				author,
+			});
+
+			Ok(())
+		}
```

Comment:
Need to add a couple of more pallet calls with a CouncilOrigin. 
- Remove an entity and its aggregate score from storage
- Remove the rating entries once an entity aggregate is removed. 
This can be a separate PR after the merge. @NiranjanAP fyi

Reply:
Noted @smohan-dw. As we talked about, we will take this one in the Next PR @NiranjanAP.